libyul/backends/wasm/TextTransform.cpp

```diff
@@ -101,6 +101,8 @@ string TextTransform::operator()(wasm::Literal const& _literal)
 
 string TextTransform::operator()(wasm::StringLiteral const& _literal)
 {
+	// StringLiteral is a special AST element used for certain builtins.
+	// The output of this will not be valid WebAssembly.
```

Comment:
I think it would be nice if this was a bit more concrete. I was wondering why we would want to insert a built-in that makes the output invalid and only after looking it up I realized that you must be thinking about the `linkersymbol` here :) I think you should outright say that a linker symbol is one (and maybe even the only?) use for this and that output should be valid after some postprocessing.

Reply:
`datasize`, `dataoffset`, and `datacopy` are the ones currently handled properly in BinaryTransform and are just included in TextTransform as `(datasize "obj")` which is invalid wasm text.

Reply:
Oh, ok then. Still I think having an example (or a hint why it's fine for it to be invalid here) would be nice.

Reply:
It is not really good, ideally we would replace it with some `i32.const` and an in-place comment `(* comment here *)`, but that means a lot more changes, and it doesn't really give any benefit for us currently.